Source/DataModels/Components/UI/ComponentVideo.h

```diff
@@ -0,0 +1,97 @@
+#pragma once
+#include "Auxiliar/Generics/Drawable.h"
+#include "DataModels/Components/Component.h"
+#include "Resources/ResourceVideo.h"
+
+struct SwsContext;
+
+class ComponentVideo : public Component, public Drawable
+{
+public:
+	ComponentVideo(bool active, GameObject* owner);
+	~ComponentVideo() override;
+	ComponentVideo(const ComponentVideo& toCopy);
+	void SetLoop(bool loop);
+	bool GetLoop();
+	bool GetCanBeRotate();
+	void SetRotateVertical(bool rotateVertical);
+	bool GetRotateVertical();
+	void SetVideo(const std::shared_ptr<ResourceVideo>& video);
+	void SetPlayAtStart(bool playAtStart);
+	bool GetPlayAtStart();
+	std::shared_ptr<ResourceVideo> GetVideo() const;
+	void SetNativeVideoFrameSize();
+	void UpdateVideoFrame();
+	void Draw() const override;
+	void Play();
+	void Pause();
+	void Stop();
+	void RestartVideo();
+	bool isPlayed();
+	bool isPlayAtStart();
```

Comment:
```suggestion
	bool IsPlayed();
	bool IsPlayAtStart();
```
Also, I have no idea what "is played" means. Whether it's done playing? I think we can find a better name.